shared/js/content-scripts/block-cookie.js

```diff
@@ -28,72 +28,109 @@
         document.__defineGetter__('cookie', () => '')
     }
 
-    function enforceCookieExpiry () {
-        var cookieSetter = document.__lookupSetter__('cookie')
-        // Enforce 7 day max expiry
-        document.__defineSetter__('cookie', function (value) {
-            const nextWeek = 604800
-            var comps = value.split(';')
-            var expiryFound = false
-            for (var i = 0; i < comps.length; i++) {
-                if (comps[i].toLowerCase().contains('session')) {
-                    expiryFound = true
-                    break
-                }
-
-                if (comps[i].trim().startsWith('expires')) {
-                    expiryFound = true
-                    const dateStr = comps[i].trim().replace('expires=', '').trim()
-                    const setDate = new Date(dateStr)
-                    const nextWeekDate = new Date()
-                    nextWeekDate.setTime(nextWeekDate.getTime() + nextWeek)
-                    if (setDate > nextWeekDate) {
-                        comps[i] = 'max-age=' + nextWeek
-                    }
+    /**
+     * Apply an expiry policy to cookies set via document.cookie.
+     * @param {string} secret Used to detect messages sent from the extension content-script
+     */
+    function applyCookieExpiryPolicy (secret) {
+        const cookieSetter = document.__lookupSetter__('cookie')
+        const cookieGetter = document.__lookupGetter__('cookie')
+        const lineTest = /(\()?(http[^)]+):[0-9]+:[0-9]+(\))?/
 
-                    break
-                } else if (comps[i].trim().startsWith('max-age')) {
-                    expiryFound = true
-                    const ageStr = comps[i].trim().replace('max-age=', '').trim()
-                    const setDate = Number(ageStr)
-                    if (setDate > nextWeek) {
-                        comps[i] = 'max-age=' + nextWeek
-                    }
-
-                    break
+        // Listen for a message from the content script which will configure the policy for this context
+        const loadPolicy = new Promise((resolve) => {
+            const messageListener = (event) => {
+                if (event && event.isTrusted && event.data && event.data.source === secret) {
+                    resolve(event.data)
+                    window.removeEventListener('message', messageListener)
                 }
             }
-
-            if (!expiryFound) {
-                comps.push('max-age=' + nextWeek)
+            window.addEventListener('message', messageListener)
+        })
+        console.log('inject cookie test', cookieSetter, secret)
```

Comment:
we probably want to drop logging before merging or put it behind a flag (there are bunch of console.log's and console.warn's in the file)